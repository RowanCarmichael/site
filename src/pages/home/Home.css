.container {
  width: 100vw;
  height: 60vh;
  background: black;
}

.background {
  height: 100%;
  left: 0;
  top: 0;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.backgroundImage {
  object-fit: cover;
  height: 100vh;
  width: 100vw;
  animation: fade-in 0.4s;
}

.content {
  display: flex;
  align-items: center;
  flex-direction: column;
  width: 100%;
  z-index: 1;
}

.header {
  color: white;
  font-size: 8rem;
  margin: 10rem 0 0 0;
  font-weight: 100;
  line-height: 8rem;
  text-align: center;
}

.divider {
  width: 48rem;
  height: 0.2rem;
  background: white;
}

.subHeader {
  color: white;
  font-size: 3.2rem;
  line-height: 4rem;
  font-weight: 100;
  text-align: center;
}

.icon {
  min-width: 14rem;
  min-height: 14rem;
}

.summaryHeader {
  margin: 10rem 4rem;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
}

.summaryHeader h1 {
  font-size: 10rem;
  font-weight: 100;
  line-height: 9rem;
  margin: 0;
  text-align: center;
}

.summary {
  margin: 10rem 4rem;
}

.summary > div {
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  justify-content: center;
  margin-bottom: 6rem;
  flex-wrap: wrap;
}

.summary > div > div {
  max-width: 80rem;
}

.summary h2 {
  font-size: 4.8rem;
  font-weight: 100;
  margin: 0;
  text-align: center;
}

.summary p {
  font-size: 2.6rem;
  font-weight: 300;
  margin: 0;
  text-align: center;
}

.link, .link:visited, .link:active {
  font-weight: bold;
  color: black;
  cursor: pointer;
  text-decoration: none;
  outline: 0;
}

@keyframes fade-in {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
