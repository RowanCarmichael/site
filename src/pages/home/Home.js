import React from 'react';
import { Container, Background } from 'react-fixed-background';
import Typewriter from '../../components/typewriter/Typewriter';
import styles from './Home.css';
import laptopBackground from '../../assets/images/laptop.jpg';
import face from '../../assets/images/face.png';
import resume from '../../assets/files/RowanCarmichaelResume.pdf';
import KeyboardIcon from '../../assets/icons/keyboard.svg';
import DevicesIcon from '../../assets/icons/devices.svg';
import ChatIcon from '../../assets/icons/chat.svg';
import TestIcon from '../../assets/icons/test.svg';
import BookIcon from '../../assets/icons/book.svg';

class Home extends React.Component {
  state = {
    loaded: false,
  }

  onLoadImage = () => {
    this.setState({ loaded: true });
  }

  render() {
    return (
      <div>
        <Container className={styles.container}>
          <img onLoad={this.onLoadImage} className={this.state.loaded ? styles.backgroundImage : null} src={laptopBackground} alt="laptopBackground" />
          <Background bringToFront className={styles.background}>
            <div className={styles.content}>
              <h1 className={styles.header}>Rowan Carmichael</h1>
              <div className={styles.divider} />
              <Typewriter speed={60} delay={3000} className={styles.subHeader} text="Frontend Engineer" />
            </div>
          </Background>
        </Container>
        <div className={styles.summaryHeader}>
          <img width="180px" src={face} alt="face" />
          <h1>What I&apos;m about</h1>
        </div>
        <div className={styles.summary}>
          <div>
            <div>
              <h2>Software Development</h2>
              <p>I&apos;m a frontend engineer passionate about React.</p>
              <p>Check out the <a target="_blank" rel="noreferrer noopener" className={styles.link} href="http://www.github.com/RowanCarmichael/site">code for this page</a> and more on my <a target="_blank" rel="noreferrer noopener" className={styles.link} href="http://www.github.com/RowanCarmichael">GitHub</a></p>
            </div>
            <KeyboardIcon className={styles.icon} />
          </div>
          <div>
            <DevicesIcon className={styles.icon} />
            <div>
              <h2>Practical Solutions</h2>
              <p>I always keep the customer in mind and will strive for the best usability experience. My day-to-day involves working closely with other frontend and backend devs, UI/UX designers, product managers and testers in a high-performing team environment</p>
            </div>
          </div>
          <div>
            <div>
              <h2>Agile Practitioner</h2>
              <p>Well versed in utilising both Scrum and Kanban agile development methodologies including experience as a Scrum (Kanban) master</p>
            </div>
            <ChatIcon className={styles.icon} />
          </div>
          <div>
            <TestIcon className={styles.icon} />
            <div>
              <h2>Automative Testing</h2>
              <p>I take great pride in code quality and fully understand the importance of creating automated tests spanning all levels of the testing pyramid to ensure the highest quality of products are released</p>
            </div>
          </div>
          <div>
            <div>
              <h2>Learner</h2>
              <p>But above all else I love to be challenged, to learn and to grow!</p>
              <p>If you&apos;d like to see more of what I can do take a look at my <a target="_blank" rel="noreferrer noopener" className={styles.link} href={resume}>Resume</a></p>
              <p>Or if you&apos;d prefer to chat, feel free to contact me via <a className={styles.link} href="mailto:carmichaelr@hotmail.co.uk">Email</a> or <a target="_blank" rel="noreferrer noopener" className={styles.link} href="http://www.linkedin.com/in/rowan-carmichael/">LinkedIn</a></p>
            </div>
            <BookIcon className={styles.icon} />
          </div>
        </div>
      </div>
    );
  }
}

export default Home;
