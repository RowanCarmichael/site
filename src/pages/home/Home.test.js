import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Home';

describe('Home page', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(<Home />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
