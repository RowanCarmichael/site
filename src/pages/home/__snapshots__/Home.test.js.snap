// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders correctly 1`] = `
<div>
  <div
    className="Container__container___1UbtE"
  >
    <img
      alt="laptopBackground"
      className={null}
      onLoad={[Function]}
      src="laptop.jpg"
    />
    <div
      className="Background__backgroundOuter___2XEdG Background__foreground___2JmNj"
    >
      <div
        className="Background__backgroundInner___34S8e"
      >
        <div
          className={undefined}
        >
          <h1
            className={undefined}
          >
            Rowan Carmichael
          </h1>
          <div
            className={undefined}
          />
          <div
            className={null}
            delay={3000}
            speed={60}
          >
            <span>
              
            </span>
            <span
              className={undefined}
            >
              |
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className={undefined}
  >
    <img
      alt="face"
      src="face.png"
      width="180px"
    />
    <h1>
      What I'm about
    </h1>
  </div>
  <div
    className={undefined}
  >
    <div>
      <div>
        <h2>
          Software Development
        </h2>
        <p>
          I'm a frontend engineer passionate about React.
        </p>
        <p>
          Check out the 
          <a
            className={undefined}
            href="http://www.github.com/RowanCarmichael/site"
            rel="noreferrer noopener"
            target="_blank"
          >
            code for this page
          </a>
           and more on my 
          <a
            className={undefined}
            href="http://www.github.com/RowanCarmichael"
            rel="noreferrer noopener"
            target="_blank"
          >
            GitHub
          </a>
        </p>
      </div>
      <keyboard.svg
        className={undefined}
      />
    </div>
    <div>
      <devices.svg
        className={undefined}
      />
      <div>
        <h2>
          Practical Solutions
        </h2>
        <p>
          I always keep the customer in mind and will strive for the best usability experience. My day-to-day involves working closely with other frontend and backend devs, UI/UX designers, product managers and testers in a high-performing team environment
        </p>
      </div>
    </div>
    <div>
      <div>
        <h2>
          Agile Practitioner
        </h2>
        <p>
          Well versed in utilising both Scrum and Kanban agile development methodologies including experience as a Scrum (Kanban) master
        </p>
      </div>
      <chat.svg
        className={undefined}
      />
    </div>
    <div>
      <test.svg
        className={undefined}
      />
      <div>
        <h2>
          Automative Testing
        </h2>
        <p>
          I take great pride in code quality and fully understand the importance of creating automated tests spanning all levels of the testing pyramid to ensure the highest quality of products are released
        </p>
      </div>
    </div>
    <div>
      <div>
        <h2>
          Learner
        </h2>
        <p>
          But above all else I love to be challenged, to learn and to grow!
        </p>
        <p>
          If you'd like to see more of what I can do take a look at my 
          <a
            className={undefined}
            href="RowanCarmichaelResume.pdf"
            rel="noreferrer noopener"
            target="_blank"
          >
            Resume
          </a>
        </p>
        <p>
          Or if you'd prefer to chat, feel free to contact me via 
          <a
            className={undefined}
            href="mailto:carmichaelr@hotmail.co.uk"
          >
            Email
          </a>
           or 
          <a
            className={undefined}
            href="http://www.linkedin.com/in/rowan-carmichael/"
            rel="noreferrer noopener"
            target="_blank"
          >
            LinkedIn
          </a>
        </p>
      </div>
      <book.svg
        className={undefined}
      />
    </div>
  </div>
</div>
`;
